File: ui/src/types/schemas/dailywire_auth.ts
import { z } from 'zod'

// 1) Device authorization initiation response
export const DeviceAuthResponseSchema = z.object({
  deviceCode: z.string(),
  userCode: z.string(),
  verificationUri: z.string(),
  verificationUriComplete: z.string().nullable().optional(),
  expiresIn: z.int().positive(),
  interval: z.int().positive(),
})
export type DeviceAuthResponse = z.infer<typeof DeviceAuthResponseSchema>

// 2) Poll response
export const PollResponseSchema = z.object({
  status: z.enum(['authorized', 'expired', 'denied']),
  message: z.string(),
})
export type PollResponse = z.infer<typeof PollResponseSchema>

// 3) Status response
export const StatusResponseSchema = z.object({
  authenticated: z.boolean(),
  containsRefreshToken: z.boolean(),
  expiresAt: z.number().nullable().optional().default(null),
})
export type StatusResponse = z.infer<typeof StatusResponseSchema>

File: server/backend/src/backend/api/models/dailywire_auth.py
from typing import Optional

from backend.api.models.base import ResponseBase, RequestBase


# Request/Response Models
class DeviceAuthResponse(ResponseBase):
    """Response from initiating device authorization flow."""
    device_code: str
    user_code: str
    verification_uri: str
    verification_uri_complete: Optional[str] = None
    expires_in: int
    interval: int


class PollRequest(RequestBase):
    """Request to poll for authorization status."""
    device_code: str


class PollResponse(ResponseBase):
    """Response from polling for authorization."""
    status: str  # "authorized", "expired", "denied"
    message: str


class StatusResponse(ResponseBase):
    """Authentication status response."""
    authenticated: bool
    contains_refresh_token: bool
    expires_at: Optional[float] = None
